### PR TITLE
198 multiple reaction watching

### PR DESCRIPTION
**Pull request by MarshallAsch**
Closes: #189, #201

---

**Describe the pull request:**

This will allow botomir to watch multiple role reaction messages and have them in different channels with different text at the same time. 

This will also add a fix to get around the limitation of only being able to watch 20 roles at time on one message.

This will also make role reactions still work if the role is renamed after the message has been created. 

Note that this will require a manual database update for all existing role reaction messages so the role reactions still work since this adds some required fields to the database. 

We also need to update the index on the reaction collection to be (guild, channel, message, emoji) so that the lookups can be faster.

---

Checklist:

- [x] Run `npm run lint` and ensure linter passes
- [x] Run `npm run test` and ensure tests pass
- [x] Verify that the changes work as expected on Discord
- [x] Update documentation / not applicable
- [x] Update changelog / not applicable

### CODE REVIEW

**Comment by MarshallAsch** [comment]
@Kayeris I think this should solve your problem and make it so you can also have the pronoun reaction message. 

@SoorajModi **DO NOT MERGE THIS ONCE IT IS APPROVED** There are some breaking changes in this that will require some manual database changes.

File: bot/commands/admin/createRoleWatch.js
Comment (SoorajModi) [inline bot/commands/admin/createRoleWatch.js:16]: We should be consistent with naming here and call our function `createRoleWatch` or rename the file `createRoleReaction.js`. However this may result in having to rename a whole bunch of things so no worries if its too much work. Or is there a reason you chose to use roleWatch instead of roleReactions in some of your files?

File: bot/commands/admin/listRoleWatch.js
Comment (SoorajModi) [inline bot/commands/admin/listRoleWatch.js:39]: In the future it might be nice to have embedded messages with details about the message instead of just the link. Dont worry about improving it for this PR

**Review by SoorajModi — CHANGES_REQUESTED** [review]
Couple more things:
- What do you think about adding and `index.js` file under `models/`, that exports all of the models. Could cleanup some of our import statements
- I think the `admin` folder is getting kind of big, what do you think about putting all of these files under a `admin/roleWatch` folder?

**Comment by SoorajModi** [comment]
Overall though, it looks really good and the implementation is great

File: bot/commands/admin/listRoleWatch.js
Comment (MarshallAsch) [reply bot/commands/admin/listRoleWatch.js:39]: I think there is a limit of 1 embed per message and I would like to play around with that formatting later, or send an embed like the link event handler. I think Id like to leave that as a separate issue because it also impacts 2 other commands

File: bot/commands/admin/createRoleWatch.js
Comment (MarshallAsch) [reply bot/commands/admin/createRoleWatch.js:16]: ya my naming has been fairly arbitrary, I dont quite like either of the names, if you have a preference or can think of anything else Id be open to changing it to that

**Comment by MarshallAsch** [comment]
> Couple more things:
> 
> * What do you think about adding and `index.js` file under `models/`, that exports all of the models. Could cleanup some of our import statements
> * I think the `admin` folder is getting kind of big, what do you think about putting all of these files under a `admin/roleWatch` folder?

I agree the folder is getting large, I have an idea about how to separate it, into smaller top level groups and Im going to make an issue for that after this is done (so that it can happen as a file move rather than delete and re adding)

File: bot/commands/admin/createRoleWatch.js
Comment (SoorajModi) [reply bot/commands/admin/createRoleWatch.js]: Okay lets address all that stuff in a separate PR then

**Review by SoorajModi — APPROVED** [review]

**Pull request merged**
